prune/maxcount.go

```diff
@@ -16,24 +16,25 @@ package prune
 
 import (
 	"context"
+	"fmt"
 	"time"
 )
 
 // pruneByMaxCount looks for and prunes resources, currently jobs and pods,
 // that exceed a user specified count (e.g. 3), the oldest resources
-// are pruned
-func pruneByMaxCount(ctx context.Context, config Config, resources []ResourceInfo) (err error) {
+// are pruned, resources to remove are returned
+func pruneByMaxCount(_ context.Context, config Config, resources []ResourceInfo) (resourcesToRemove []ResourceInfo, err error) {
 	config.log.V(1).Info("pruneByMaxCount running ", "max count", config.Strategy.MaxCountSetting, "resource count", len(resources))
+	if config.Strategy.MaxCountSetting < 0 {
+		return resourcesToRemove, fmt.Errorf("max count setting less than zero")
+	}
 
 	if len(resources) > config.Strategy.MaxCountSetting {
```

Comment:
yeah, this is perhaps misleading in the name 'maxcount', what this means is 'keep this many resources', so if you had 10 resources on the system, and this value is set to 5, it would leave a max of 5 resources on the system.